packager/certfetcher/certfetcher.go

```diff
@@ -159,6 +130,52 @@ func New(email string, eabKid string, eabHmac string, certSignRequest *x509.Cert
 	}, nil
 }
 
+func setupChallenges(client *lego.Client, httpChallengePort int,
+	httpChallengeWebRoot string, tlsChallengePort int,
+	dnsProvider string) (*lego.Client, error) {
```

Comment:
Either:
1. Don't take a `client`; do the `lego.NewClient` call in here.
2. Just return `error` and add a function docstring that it mutates `client` in-place.

I lean weakly towards 1 since it eliminates a bit more duplication.

Reply:
Done.